runtime/protocol/http-client-engines/http-client-engine-okhttp/jvm/src/aws/smithy/kotlin/runtime/http/engine/okhttp/HttpEngineEventListener.kt

```diff
@@ -158,14 +161,22 @@ internal class HttpEngineEventListener(
 
     override fun requestBodyStart(call: Call) = trace { "sending request body" }
 
-    override fun requestBodyEnd(call: Call, byteCount: Long) =
+    override fun requestBodyEnd(call: Call, byteCount: Long) {
+        requestTimeEnd = TimeSource.Monotonic.markNow()
         trace { "finished sending request body: bytesSent=$byteCount" }
+    }
 
     override fun requestFailed(call: Call, ioe: IOException) = trace(ioe) { "request failed" }
 
     override fun requestHeadersStart(call: Call) = trace { "sending request headers" }
 
-    override fun requestHeadersEnd(call: Call, request: Request) = trace { "finished sending request headers" }
+    override fun requestHeadersEnd(call: Call, request: Request) {
+        if (request.body == null) {
```

Comment:
I think you're right, I just missed it in first review

Reply:
Yes what confused me is that this entire file has the events out-of-order (i.e `requestHeadersEnd` [is defined](https://github.com/awslabs/smithy-kotlin/blob/service-benchmarks/runtime/protocol/http-client-engines/http-client-engine-okhttp/jvm/src/aws/smithy/kotlin/runtime/http/engine/okhttp/HttpEngineEventListener.kt#L173) **after** [`requestBodyStart`](https://github.com/awslabs/smithy-kotlin/blob/service-benchmarks/runtime/protocol/http-client-engines/http-client-engine-okhttp/jvm/src/aws/smithy/kotlin/runtime/http/engine/okhttp/HttpEngineEventListener.kt#L162) even though it occurs before in the  chain of events). It could be worth reordering these to match the order [shown in the OkHttp docs](https://square.github.io/okhttp/features/events/#eventlistener).

Reply:
Agreed, that would make it easier to parse this file. I can submit a mini-PR after this one gets merged.